### PR TITLE
feat: Allow owner to share workflows/credentials they don't own (no-changelog)

### PR DESCRIPTION
**Pull request by valya**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Comment by valya** [comment]
Yes, this is the backend for it

File: packages/cli/test/integration/credentials.ee.test.ts [hunk 1]
```diff
@@ -406,14 +410,49 @@ describe('PUT /credentials/:id/share', () => {
...
+		});
+		expect(sharedCredentials).toHaveLength(2);
+	});
+
+	test('should respond 403 for non-owned credentials for non-shared members', async () => {
+		const savedCredential = await saveCredential(randomCredentialPayload(), { user: member });
+
+		const response = await authAnotherMemberAgent
+			.put(`/credentials/${savedCredential.id}/share`)
+			.send({ shareWithIds: [anotherMember.id] });
```
Comment (ivov) [inline packages/cli/test/integration/credentials.ee.test.ts:434]: I expect we want to test a non-sharee sharing a non-owned cred with someone other than themselves? Same in other suite.

**Comment by ivov** [comment]
Confirmed working with workflows and credentials.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.20.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.20.0)

**Pull request merged**